src/main/java/com/jlr/ttl/ds/api/services/VehicleService.java

```diff
@@ -1,63 +1,72 @@
-        this.vehicleRepository = vehicleRepository;
-    }
-
-    public List<Vehicle> getAllVehiclesService() {
-        return vehicleRepository.findAll();
-    }
+    /**
+     * Fetch the vehicle by the id passed
+     *
+     * @return Vehicle
+     * @since v1
+     */
 
-    public void addNewVehicleService(Vehicle vehicle) {
-        Optional<Vehicle> vehicleByVin = vehicleRepository.findById(vehicle.getVin().toString());
-        if (vehicleByVin.isPresent()) {
-            throw new IllegalStateException();
+    @TrackExecutionTime
+    public VehicleResponse getVehicleByID(String id) throws ServiceBusinessException {
+        if(id==null || id.length()==0){
+            String errorMessage = "No vehicle id was provided";
+            log.warn(errorMessage);
+            throw new IllegalArgumentException(errorMessage);
+        }
+        try {
+            DSTableInterface<Vehicle> dbResponse = vehicleRepository.findById(id)
+                    .orElseThrow(() -> new VehicleNotFoundException("No vehicle with ID : " + id + " was found"));
+            return VehicleValueMapper.entityToResponse(dbResponse.createEntity());
+        }catch (VehicleNotFoundException vehicleNotFoundException){
+            String errorMessage = vehicleNotFoundException.getMessage();
+            log.warn(errorMessage);
+            throw new VehicleNotFoundException(errorMessage);
+        }catch (Exception ex) {
+            throw new ServiceBusinessException("Exception occurred while fetching vehicle with id : " + id);
         }
-        vehicleRepository.save(vehicle);
-    }
-
-    public void deleteVehicleService(String vin) {
-       vehicleRepository.findById(vin)
-                .orElseThrow(IllegalStateException::new);
-       vehicleRepository.deleteById(vin);
-    }
-
-    public Vehicle findVehicleByVinService(String vin) {
-        return vehicleRepository.findById(vin)
-                .orElseThrow(IllegalStateException::new);
-    }
-
-    public List<Vehicle> findVehiclesByLocService(String loc) {
-        List<Vehicle> vehicles = new ArrayList<>();
-        vehicles = vehicleRepository.findVehiclesByLoc(loc);
-        // TODO: Validate loc from locations table
-//        if () {
-//            throw new IllegalStateException();
-//        }
-        return vehicles;
     }
 
-    public void updateVehicleService(String vin, Vehicle vehicle) {
-        Vehicle existingVehicle = vehicleRepository.findById(vin)
-                .orElseThrow(IllegalStateException::new);
+    /**
+     * Getting all the vehicles from database.
+     * @return a list of VehicleResponse
+     */
+    public List<VehicleResponse> getAllVehicles() throws ServiceBusinessException {
+        try {
+            List<VehiclesTable> dbResponse = vehicleRepository.findAll();
 
-        existingVehicle.setLoc_code(vehicle.getLoc_code());
-        vehicleRepository.save(existingVehicle);
+            List<VehicleResponse> finalList = new ArrayList<VehicleResponse>();
```

Comment:
Refractor this variable to have a more descriptive name such as listOfVehicleResponse